### PR TITLE
frontend: Fix various library crashes and empty library items

### LINKED ISSUE #2354: [Bug]: Bottles freezes after every launch after the last update
### Describe the bug

obraz
Bottles is either stuck on "Downloading 1/3 components" or show the above icon and immidietly freezes after the last update.

### To Reproduce

1. Open bottles
2. Wait for it to launch
3. Freezes, can't even move the windows around

### Package

Flatpak from Flathub

### Distribution

OpenSUSE microos

### Troubleshooting Logs

```shell
Version: 2022.11.14
Display:
    X.org: true
    X.org (port): :0
    Wayland: true
Graphics:
    vendors:
        nvidia:
            vendor: nvidia
            envs:
                __NV_PRIME_RENDER_OFFLOAD: '1'
                __GLX_VENDOR_LIBRARY_NAME: nvidia
                __VK_LAYER_NV_optimus: NVIDIA_only
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json:/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 6.0.10-1-default
Distro:
    Name: GNOME
    Version: '43'
Disk:
    Total: 16772526080
    Free: 16772370432
RAM:
    MemTotal: "33,5\u202FGB"
    MemAvailable: "28,0\u202FGB"
Bottles_envs: null

```

### Display Protocol

Wayland

### Display Backend used by Bottles

Wayland

### Additional context

I'm currently using a hotspot from my phone which gnome software detects as a "measured connection", not sure if this has anything to do with it though

### LINKED ISSUE #2386: [Bug]: Deleting a bottle does not remove its library entries, causing Bottles to not launch
### Describe the bug

When a bottle is deleted, the library entries for its installed apps are not removed from Bottles. Bottles will not start after relaunching, with the following output:

```python
23:42:07 (INFO) Library saved 
Traceback (most recent call last):
  File "/app/share/bottles/bottles/frontend/windows/main_window.py", line 155, in set_manager
    self.page_library = LibraryView(self)
  File "/app/share/bottles/bottles/frontend/views/library.py", line 43, in __init__
    self.update()
  File "/app/share/bottles/bottles/frontend/views/library.py", line 56, in update
    entry = LibraryEntry(self, u, e)
  File "/app/share/bottles/bottles/frontend/widgets/library.py", line 62, in __init__
    self.__remove_entry()
  File "/app/share/bottles/bottles/frontend/widgets/library.py", line 137, in __remove_entry
    self.library.remove_entry(self)
  File "/app/share/bottles/bottles/frontend/views/library.py", line 66, in remove_entry
    entry.hide()
  File "/app/share/bottles/bottles/frontend/widgets/library.py", line 169, in hide
    self.get_parent().set_visible(False)
AttributeError: 'NoneType' object has no attribute 'set_visible'
```

To fix the issue and start Bottles again, the problematic library entry in ~/.var/app/com.usebottles.bottles/data/bottles/library.yml must be removed. 

### To Reproduce

1. Create a new bottle
2. Add a shortcut to an app
3. Add the app to library
4. Delete the bottle
5. The library entry is not removed, and relaunching Bottles will cause it to not start. 

### Package

Flatpak from Flathub

### Distribution

Arch Linux

### Troubleshooting Logs

```shell
Official Package: true
Version: 2022.12.14
Display:
    X.org: true
    X.org (port): :0
    Wayland: false
Graphics:
    vendors:
        amd:
            vendor: amd
            envs:
                DRI_PRIME: '1'
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/radeon_icd.x86_64.json:/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/radeon_icd.i686.json
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 6.0.12-arch1-1
Distro:
    Name: GNOME
    Version: '43'
Disk:
    Total: 8314679296
    Free: 8314531840
RAM:
    MemTotal: 15.5GiB
    MemAvailable: 9.9GiB
Bottles_envs: null
```

### Display Protocol

X11

### Display Backend used by Bottles

X11

### Additional context

_No response_

### LINKED ISSUE #2416: [Bug]: Bottles stuck/broken on Starting Up
### Describe the bug

After installing bottles software from flathub bottles is stuck on Starting Up... 3/3 screen
I have a gigabit internet so thats not a problem, i tried to remove the network access but it still wont work

### To Reproduce

Start bottles

### Package

Flatpak from Flathub

### Distribution

Fedora 37

### Troubleshooting Logs

```shell
As bottles wont start there is no way to provide this data, but im on the latest Fedora 37 Kernel/Update using Wayland with an AMD CPU/GPU system
```

### Additional context

_No response_

### LINKED ISSUE #2490: [Bug]: Renaming a bottle breaks library entries from it
### Describe the bug

When preparing screenshots for earlier post I renamed my bottle and programs to something more typical and when I restarted Bottles all entries from that bottle were broken. Hovering them with mouse doesn't show any UI so you're just left with empty covers that you can't do anything with. I opened the library.yml and noticed that it wasn't updated at all. The "name" and "path" values under "bottle" value had the old bottle name before change. Program "name" value also wasn't changed. When I changed the "bottle":"name" value to the new bottle name and restarted the Bottles the library entry started working again. Then I also changed the program name in library.yml so it shows the new name.

### To Reproduce

1. Make a new bottle with some program that Bottles downloads the cover for (for example Notepad++)
2. Install Notepad++
3. Add Notepad++ to library
4. Rename bottle with Notepad++ to a different name
5. Restart Bottles
6. Go to library and hover mouse over the empty cover

### Package

Flatpak from Flathub

### Distribution

Linux Mint 21.1 Cinnamon

### Troubleshooting Logs

```shell
Official Package: true
Version: 2022.12.14.1
Display:
    X.org: true
    X.org (port): :99.0
    Wayland: false
Graphics:
    vendors:
        amd:
            vendor: amd
            envs:
                DRI_PRIME: '1'
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/radeon_icd.x86_64.json:/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/radeon_icd.i686.json
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 5.15.0-56-generic
Distro:
    Name: GNOME
    Version: '43'
Disk:
    Total: 8356999168
    Free: 8356851712
RAM:
    MemTotal: 15.6GiB
    MemAvailable: 12.5GiB
Bottles_envs: null
```

### Additional context

_No response_

### LINKED ISSUE #2511: [Bug]: Empty library items
### Describe the bug

Also, no way to delete those items.

See image below.

Screenshot from 2023-01-10 19-31-16

### To Reproduce

1.  does not apply

### Package

Flatpak from Flathub

### Distribution

Ubuntu 22.04.1 LTS

### Troubleshooting Logs

```shell
Official Package: true
Version: 2022.12.14.1
Display:
    X.org: true
    X.org (port): :99.0
    Wayland: false
Graphics:
    vendors:
        nvidia:
            vendor: nvidia
            envs:
                __NV_PRIME_RENDER_OFFLOAD: '1'
                __GLX_VENDOR_LIBRARY_NAME: nvidia
                __VK_LAYER_NV_optimus: NVIDIA_only
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json:/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 5.15.0-57-generic
Distro:
    Name: GNOME
    Version: '43'
Disk:
    Total: 8353746944
    Free: 8353583104
RAM:
    MemTotal: 15.6GiB
    MemAvailable: 12.3GiB
Bottles_envs: null
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kinsteen**
# Description
- if the bottle was deleted but the library entry remains, delete it
- if images are missing, redownload them if they were present
- deleting a bottle now deletes the library entry to avoid empty grey entries
- Renaming a program renames the library entry and redownloads a cover

---

- Fixes #2416 
- Fixes #2354
- Fixes #2511
- Fixes #2490
- Fixes #2386

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce.
- [X] locally, with multiple programs that have thumbnails or not

### CODE REVIEW

File: bottles/frontend/widgets/library.py [hunk 1]
```diff
@@ -74,6 +79,15 @@ def __init__(self, library, uuid, entry, *args, **kwargs):
 
         if entry.get('thumbnail'):
             path = ThumbnailManager.get_path(self.config, entry['thumbnail'])
+
+            if path is None:
+                # redownloading *should* never fail as it was successfully downloaded before
```
Comment (mirkobrombin) [inline bottles/frontend/widgets/library.py:84]: do we assume that the connection is accessible?

**Review by mirkobrombin — COMMENTED** [review]
looks correct, I left a comment

File: bottles/frontend/widgets/library.py (hunk 1 above)
Comment (Kinsteen) [reply bottles/frontend/widgets/library.py:84]: I did make this assumption... Will change it!

**Comment by Kinsteen** [comment]
Did the requested change, should be ready now

**Review by mirkobrombin — APPROVED** [review]

**Pull request merged**